marlowe-playground-client/src/SimulationPage/View.purs

```diff
@@ -0,0 +1,564 @@
+        come back later on.
+      , article [ class_ (ClassName "documentation-panel") ]
+          (toHTML (state ^. _helpContext))
+      -}
+------------------------------------------------------------
+startSimulationWidget :: forall p. Slot -> HTML p Action
+startSimulationWidget initialSlot =
+  cardWidget "Simulation has not started yet"
+    $ div [ classes [] ]
+        [ div [ classes [ ClassName "slot-input", ClassName "initial-slot-input" ] ]
+            [ spanText "Initial slot:"
+            , marloweActionInput true (SetInitialSlot <<< wrap) initialSlot
+            ]
+        , div [ classes [ ClassName "transaction-btns", flex, justifyCenter ] ]
+            [ button
+                [ classes [ btn, bold ]
+                , onClick $ const $ Just StartSimulation
+                ]
+                [ text "Start simulation" ]
+            ]
+        ]
+
+------------------------------------------------------------
+simulationStateWidget ::
+  forall p.
+  State ->
+  HTML p Action
+simulationStateWidget state =
+  let
+    currentSlot = state ^. (_currentMarloweState <<< _executionState <<< _SimulationRunning <<< _slot <<< to show)
+
+    expirationSlot = state ^. (_marloweState <<< _Head <<< _contract <<< to contractMaxTime)
+
+    contractMaxTime = case _ of
+      Nothing -> "Closed"
+      Just contract ->
+        let
+          t = (_.maxTime <<< unwrap <<< timeouts) contract
+        in
+          if t == zero then "Closed" else show t
+
+    indicator name value =
+      div_
+        [ span
+            [ class_ bold ]
+            [ text $ name <> ": " ]
+        , span_ [ text value ]
+        ]
+  in
+    div
+      [ classes [ flex, justifyBetween ] ]
+      [ indicator "current slot" currentSlot
+      , indicator "expiration slot" expirationSlot
+      ]
+
+------------------------------------------------------------
+actionWidget ::
+  forall p.
+  State ->
+  HTML p Action
+actionWidget state =
+  cardWidget "Actions"
+    $ div [ classes [] ]
+        [ ul [ class_ (ClassName "participants") ]
+            if (Map.isEmpty possibleActions) then
+              [ text "No valid inputs can be added to the transaction" ]
+            else
+              (actionsForParties possibleActions)
+        , div [ classes [ ClassName "transaction-btns", flex, justifyCenter ] ]
+            [ button
+                [ classes [ btn, bold, (Classes.disabled $ not isEnabled), spaceRight ]
+                , disabled $ not $ hasHistory state
+                , onClick $ const $ Just Undo
+                ]
+                [ text "Undo" ]
+            , button
+                [ classes [ btn, bold, (Classes.disabled $ not isEnabled) ]
+                , onClick $ const $ Just ResetSimulator
+                ]
+                [ text "Reset" ]
+            ]
+        ]
+  where
+  -- FIXME: I think the contract should always be valid if we are in the simulation
```

Comment:
indeed it should

Reply:
removed all code that askes if it was valid